docs/concepts/configuration/pod-priority-preemption.md

```diff
@@ -0,0 +1,177 @@
+change priority of existing pods. The value of such PriorityClass will be used only
+for pods created after the PriorityClass is added.
+
+#### Example PriorityClass
+```yaml
+apiVersion: v1
+kind: PriorityClass
+metadata:
+  name: high-priority
+value: 1000000
+globalDefault: false
+description: "This priority class should be used for XYZ service pods only."
+```
+
+## Pod Priority
+Once you have one or more PriorityClasses, you can create pods which specify one
+of those PriorityClass names in their spec.
+The following YAML is an example of a pod configuration that uses the PriorityClass
+created above. Priority admission controller checks the spec and resolves the
+priority of the pod to 1,000,000.
+
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+  labels:
+    env: test
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    imagePullPolicy: IfNotPresent
+  priorityClassName: high-priority
+```
+
+## Preemption
+When pods are created, they go to a queue and wait to be scheduled. Scheduler picks a pod
+from the queue and tries to schedule it on a node. If no node is found that satisfies
+all the specified requirements of the pod, the pod is determined infeasible. At this
+point preemption logic is triggered for the pending pod. Let's call the pending pod P.
+Preemption logic tries to find a node where removal of pods with lower priority than
+P helps schedule P. If such node is found, one or more lower priority pods will
+be deleted from the node. Once the pods are gone, P may be scheduled on the node. 
+
+### Limitations of Preemption (alpha version)
+
+#### Starvation of Preempting Pod
+When pods are preempted, the victims get their
+[graceful termination period](https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods).
+They have so much time to finish their work and exit. If they don't, they will be
+killed. This graceful termination period creates a time gap between the point that
+scheduler preempts pods until the pending pod (P) can be scheduled on the node (N).
+When there are multiple victims on node N, they may exit or get terminated at
+various points in time. When one exits, it creates some room on node N. Pod P can
+only be scheduled on node N when __all__ the victims exit, but other pods with different requirements
+may exist in the scheduling queue that fit on node N when some of the victims have
+exited. Scheduler may schedule them on the node. In such a case, it is likely that
+when all victims exit, pod P won't fit on node N anymore. So, scheduler will have to
+preempt other pods on node N or another node to let P schedule. This scenario may 
+be repeated again for the second and subsequent rounds of preemption and P may not
+get scheduled for a while. This scenario can cause problems in various clusters, but
+is particularly problematic in clusters where many new pods are created all the time.
+
+We intend to address this problem in beta version of pod preemption. The solution
```

Comment:
Let's be very explici here: "Fixing this is a Beta blocker".

@davidopp - do you agree?

Reply:
I agree that it is a beta blocker, but should we put it in a user doc?
I am going to write a separate design doc on it. I will put the milestones there.

Reply:
Race condition in fixing and commenting.
I changed it to we will address in beta.